### PR TITLE
Added Strict typing in fields of models for oneof/anyof cases

### LINKED ISSUE #21: Strict typing is missing in fields of models in oneof/anyof cases
Following OneOf case is currently not working in JsonMapper
let two models:
**VehicleA:**
``` json
{
    "numberOfTyres": 4
    "model": 2019
}
```
**VehicleB:**
``` json
{
    "numberOfTyres": "4"
    "model": "2019"
}
```

Consider case: `oneof(VehicleA,VehicleB)` ON:
``` json
{
    "numberOfTyres": 4
    "model": 2019
}
```
Ideally, this should be mapped by type `VehicleA` but currently this is being mapped by both `VehicleA` and `VehicleB`

### PR DESCRIPTION
**Pull request by asadali214**
Closes #21

### CODE REVIEW

File: src/JsonMapper.php [hunk 1]
```diff
@@ -365,14 +365,14 @@ protected function isValueDeserializable($value, $deserializer)
...
-     * @param $className        string         Name of the class
-     * @param $forMultipleTypes bool           Should map for multiple types?
+     * @param $jvalue         mixed         Raw normalized data for the property
+     * @param $type           string        Type found by inspectProperty()
+     * @param $typeOfs        string|null   OneOf/AnyOf types hint found by
+     *                        inspectProperty in maps annotation
+     * @param $factoryMethods string[]|null Callable factory methods for property
+     * @param $namespace      string        Namespace of the class
+     * @param $className      string        Name of the class
+     * @param $strict         bool          Should map for multiple types?
```
Comment (sufyankhanrao) [inline src/JsonMapper.php:375]: Nit, Non-blocker: Description can be misleading in the future, I'd recommend using something like `Should use strict type checking`.

**Review by sufyankhanrao — APPROVED** [review]
These changes look good to me.

File: src/JsonMapper.php (hunk 1 above)
Comment (asadali214) [reply src/JsonMapper.php]: Fixed

**Review by sufyankhanrao — APPROVED** [review]
The changes are approved.

**Pull request merged**